torch/_dynamo/variables/builtin.py

```diff
@@ -1203,7 +1203,7 @@ def _grad_changed(old, new):
                                 #   -k test_dropout_backward_layout_torch_jagged_cpu
```

Comment:
I suppose this is a pre-existing problem but in line 1183, when we start looking for the base of the grad tensor, shouldn't we have *first* attempted to see if we had already had it in `input_source_to_var` as per this logic in builder?

```
        # NB: this just says we accessed a tensor from the same source again
        # (e.g., a tensor lives in a global foo, and we LOAD_GLOBAL it twice).
        # This is distinct from two distinct sources mapping to the same
        # Tensor (per id())!  No guard is necessary here.  See below for the
        # other case.
        is_duplicate_tensor = source in self.tx.output.input_source_to_var
        if is_duplicate_tensor:
            return self.tx.output.input_source_to_var[source]
```